pex/environment.py

```diff
@@ -161,6 +161,9 @@ def _resolve(self, working_set, reqs):
     # Resolve them one at a time so that we can figure out which ones we need to elide should
     # there be an interpreter incompatibility.
     for req in reqs:
+      if req.marker and not req.marker.evaluate():
+        TRACER.log('Skipping activation of `%s` due to environment marker de-selection' % req)
```

Comment:
With logging, it's preferred to do the below style, because it allows for lazy string interpolation (e.g. turning off logging won't load any of that code). Refer to https://stackoverflow.com/a/4149231.
```
TRACER.log('Skipping activation of `%s` due to environment marker de-selection', req)
```

If you have multiple things to interpolate, then it would look like:
```
TRACER.log("%s %s %s", v1, v2, v3)
```

I don't think we use this style anywhere in PEX so I understand if you wouldn't want to start introducing it now, but wanted to bring it up so we're aware.

Reply:
Thanks. I'm aware; however, `TraceLogger` is not a `logging.Logger` and does not have a lazy formatting API:
https://github.com/pantsbuild/pex/blob/c69213893bdbe7aa1648799d9d1a94753748942c/pex/tracer.py#L66